platforms/jvm/toolchains-jvm-shared/src/main/java/org/gradle/jvm/toolchain/internal/JavaToolchainQueryService.java

```diff
@@ -175,6 +176,17 @@ private JavaToolchain resolveToolchain(JavaToolchainSpecInternal requestedSpec,
         }
     }
 
+    private Set<JavaInstallationCapability> transformCapabilities(JavaToolchainSpec actualSpec, Set<JavaInstallationCapability> requiredCapabilities) {
+        if (actualSpec.getNativeImageCapable().getOrElse(false)) {
```

Comment:
The semantics of this we define here seem odd.

I would expect:
1. `nativeImageCapable=true` means I _must_ have a native image capable JDK
2. `nativeImageCapable=false` means I _must_ have a _non_ native image capable JDK
3. `nativeImageCapable=unset` means I have no preference

This current PR aliases 2 and 3 to mean `I have no preference`.

I am not sure how often someone would say they don't want a native image VM, but doing something like:

```
java {
    toolchain {
        nativeImageCapable = false
    }
}
```

and getting GraalVM seems wrong to me

Reply:
I went down that path because `native-image` is an additional thing. It does not influence the behavior of `java` or `javac`. And thus, there does not seem to be a use case for excluding it.

Reply:
Discussed this further with @jvandort and we are going to go with the semantics of this PR: `false` or unset means I don't care.
Documentation will clarify that.

If in the future, the use case of saying that the selected toolchain cannot have `native-image` appears, we can always revisit that decision.